nilearn/plotting/surface/surf_plotting.py

```diff
@@ -1011,94 +1006,27 @@ def plot_surf_roi(
 
     nilearn.surface.vol_to_surf : For info on the generation of surfaces.
     """
-    # set default view to dorsal if hemi is both and view is not set
-    check_params(locals())
-    if view is None:
-        view = "dorsal" if hemi == "both" else "lateral"
-
-    roi_map, surf_mesh, bg_map = check_surface_plotting_inputs(
-        roi_map, surf_mesh, hemi, bg_map
-    )
-
-    if engine == "matplotlib" and avg_method is None:
-        avg_method = "median"
-
-    # preload roi and mesh to determine vmin, vmax and give more useful error
-    # messages in case of wrong inputs
-    check_extensions(roi_map, DATA_EXTENSIONS, FREESURFER_DATA_EXTENSIONS)
-
-    roi = load_surf_data(roi_map)
-
-    idx_not_na = ~np.isnan(roi)
-    if vmin is None:
-        vmin = float(np.nanmin(roi))
-    if vmax is None:
-        vmax = float(1 + np.nanmax(roi))
-
-    mesh = load_surf_mesh(surf_mesh)
-
-    if roi.ndim != 1:
-        raise ValueError(
-            "roi_map can only have one dimension but has "
-            f"{roi.ndim} dimensions"
-        )
-    if roi.shape[0] != mesh.n_vertices:
-        raise ValueError(
-            "roi_map does not have the same number of vertices "
-            "as the mesh. If you have a list of indices for the "
-            "ROI you can convert them into a ROI map like this:\n"
-            "roi_map = np.zeros(n_vertices)\n"
-            "roi_map[roi_idx] = 1"
-        )
-    if (roi < 0).any():
-        # TODO raise ValueError in release 0.13
-        warn(
-            (
-                "Negative values in roi_map will no longer be allowed in"
-                " Nilearn version 0.13"
-            ),
-            DeprecationWarning,
-            stacklevel=find_stack_level(),
-        )
-    if not np.array_equal(roi[idx_not_na], roi[idx_not_na].astype(int)):
-        # TODO raise ValueError in release 0.13
-        warn(
-            (
-                "Non-integer values in roi_map will no longer be allowed in"
-                " Nilearn version 0.13"
-            ),
-            DeprecationWarning,
-            stacklevel=find_stack_level(),
-        )
-
-    if cbar_tick_format == "auto":
-        cbar_tick_format = "." if engine == "plotly" else "%i"
```

Comment:
Is there a special reason for the value of `cbar_tick_format` or it was a random choice?

In current implementation, I leave `plot_surf` to adjust this value, which sets 
for `matplotlib`:
```
cbar_tick_format = (
            "%.2g" if cbar_tick_format == "auto" else cbar_tick_format
        )
```
for `plotly`:
```
        cbar_tick_format = (
            ".1f" if cbar_tick_format == "auto" else cbar_tick_format
        )
```
 If it is convenient, I'll leave the code as is, otherwise I'll update.